### PR TITLE
[PL-25545]: ng csm add secret changes

### PR DESCRIPTION
**Pull request by shreyas-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36560)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 8 Issues 
* Vulnerability 1 Vulnerabilities 
 * Code issue 6 Code issues 
* Anti-pattern 1 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36560)

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]
> Reviewed 10 of 10 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36560)* status: 1 change requests. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java` line 124 at r2](https://reviewable.io/reviews/harness/harness-core/36560#-NA4Zh5v-mPRJh1a_Taj:-NA4Zh5v-mPRJh1a_Tak:b-35hqiw) ([raw file](https://github.com/harness/harness-core/blob/28e0cae9cf352b3eda5239378ec6edf3efff3ad3/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java#L124)):*
> ```Java
>     this.globalEncryptDecryptClient = globalEncryptDecryptClient;
>     this.ngConnectorSecretManagerService = ngConnectorSecretManagerService;
>     this.objectMapper = objectMapper;
> ```

We should use NG_DEFAULT_OBJECT_MAPPER.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java` line 162 at r2](https://reviewable.io/reviews/harness/harness-core/36560#-NA4Z86i3rZ68Iic2qwI:-NA4Z86i3rZ68Iic2qwJ:b6jczo4) ([raw file](https://github.com/harness/harness-core/blob/28e0cae9cf352b3eda5239378ec6edf3efff3ad3/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java#L162)):*
> ```Java
>         customSecretManagerConfigDTO.getProjectIdentifier(), customSecretManagerConfigDTO.getIdentifier());
>     CustomSecretManagerConnectorDTO customSecretManagerConnectorDTO =
>         (CustomSecretManagerConnectorDTO) connectorDTO.getConnectorInfo().getConnectorConfig();
> ```

Why connector config is type casted to connector dto?
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java` line 166 at r2](https://reviewable.io/reviews/harness/harness-core/36560#-NA4_4uP2QAPLV4PfWhD:-NA4_4uP2QAPLV4PfWhE:bkaytlh) ([raw file](https://github.com/harness/harness-core/blob/28e0cae9cf352b3eda5239378ec6edf3efff3ad3/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java#L166)):*
> ```Java
>       customSecretManagerConnectorDTO.getTemplate().setTemplateInputs(
>           objectMapper.readValue(encryptedData.getPath(), Map.class));
>     } catch (JsonProcessingException e) {
> ```

Rename e to exception and log actual error.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java` line 168 at r2](https://reviewable.io/reviews/harness/harness-core/36560#-NA4_8ar4dsAQ3oZi7Gb:-NA4_8ar4dsAQ3oZi7Gc:b-enw71d) ([raw file](https://github.com/harness/harness-core/blob/28e0cae9cf352b3eda5239378ec6edf3efff3ad3/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java#L168)):*
> ```Java
>     } catch (JsonProcessingException e) {
>       throw new SecretManagementException(
>           SECRET_MANAGEMENT_ERROR, "Value passed can not be processed as map. Parsing error", USER);
> ```

include actual error message from exception
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/impl/NGConnectorSecretManagerServiceImpl.java` line 69 at r2](https://reviewable.io/reviews/harness/harness-core/36560#-NA4bQHHE6teFFPD07c-:-NA4bQHHE6teFFPD07c0:b-35hqiw) ([raw file](https://github.com/harness/harness-core/blob/28e0cae9cf352b3eda5239378ec6edf3efff3ad3/440-connector-nextgen/src/main/java/io/harness/connector/impl/NGConnectorSecretManagerServiceImpl.java#L69)):*
> ```Java
>     this.customSecretManagerHelper = customSecretManagerHelper;
>     this.customEncryptorsRegistry = customEncryptorsRegistry;
>     this.objectMapper = objectMapper;
> ```

We should use NG_DEFAULT_OBJECT_MAPPER.
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/impl/NGConnectorSecretManagerServiceImpl.java` line 131 at r2](https://reviewable.io/reviews/harness/harness-core/36560#-NA4bSvl8PDE1dg7J7OU:-NA4bSvl8PDE1dg7J7OV:bb7od0r) ([raw file](https://github.com/harness/harness-core/blob/28e0cae9cf352b3eda5239378ec6edf3efff3ad3/440-connector-nextgen/src/main/java/io/harness/connector/impl/NGConnectorSecretManagerServiceImpl.java#L131)):*
> ```Java
>       customNGSecretManagerConfigDTO.getTemplate().setTemplateInputs(objectMapper.readValue(path, Map.class));
>     } catch (JsonProcessingException e) {
>       throw new RuntimeException("Can not parse the passed string to map. Invalid input");
> ```

same

File: 120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java [hunk 1]
```diff
@@ -118,17 +138,53 @@ public NGEncryptedData createSecretText(String accountIdentifier, SecretDTOV2 dt
...
+      case Inline:
+        if (isReadOnlySecretManager(secretManager)) {
+          throw new SecretManagementException(SECRET_MANAGEMENT_ERROR, READ_ONLY_SECRET_MANAGER_ERROR, USER);
+        }
+        encrypt(encryptedData, secret.getValue(), SecretManagerConfigMapper.fromDTO(secretManager));
+      case Reference:
+        validatePath(encryptedData.getPath(), encryptedData.getEncryptionType());
+      case CustomSecretManagerValues:
+        // At the time of creation, ensure secret exist.
+        validateCustomSecretManagerPathValue(accountIdentifier, encryptedData, secretManager, secret.getValue());
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java:151]: we are missing break statements here?

**Review by meenakshi-harness — CHANGES_REQUESTED** [review]

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36560)* status: 1 file(s) to review. 2 change requests. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java` line 169 at r3](https://reviewable.io/reviews/harness/harness-core/36560#-NA81Ql96b5-659XDJ_j:-NA81Ql96b5-659XDJ_k:b1ltwxv) ([raw file](https://github.com/harness/harness-core/blob/fad7db7352e65ccc9e83ebd571aeee45ed72c997/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java#L169)):*
> ```Java
>     } catch (JsonProcessingException exception) {
>       log.error("Exception when converting user input to template input. "
>           + "Exception message: " + exception.getMessage() + " Exception: " + exception);
> ```

correct way to log would be: log.error("Message here", exception);
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java` line 171 at r3](https://reviewable.io/reviews/harness/harness-core/36560#-NA81Xd8-qpLQ1HBd1lo:-NA81Xd8-qpLQ1HBd1lp:bppjbk4) ([raw file](https://github.com/harness/harness-core/blob/fad7db7352e65ccc9e83ebd571aeee45ed72c997/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java#L171)):*
> ```Java
>           + "Exception message: " + exception.getMessage() + " Exception: " + exception);
>       throw new SecretManagementException(SECRET_MANAGEMENT_ERROR,
>           "Value passed can not be processed as map. Parsing error : " + exception.getMessage(), USER);
> ```

Use String.format instead of string concatenation.
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/impl/NGConnectorSecretManagerServiceImpl.java` line 134 at r3](https://reviewable.io/reviews/harness/harness-core/36560#-NA816BO1pt_xAKUZ8Wx:-NA816BO1pt_xAKUZ8Wy:bppjbk4) ([raw file](https://github.com/harness/harness-core/blob/fad7db7352e65ccc9e83ebd571aeee45ed72c997/440-connector-nextgen/src/main/java/io/harness/connector/impl/NGConnectorSecretManagerServiceImpl.java#L134)):*
> ```Java
>           + "Exception message: " + exception.getMessage() + " Exception: " + exception);
>       throw new RuntimeException(
>           "Can not parse the passed string to map. Invalid input: " + path + " exception: " + exception.getMessage());
> ```

Use String.format instead of string concatenation.

**Pull request closed without merging**